BookingsAPI/Bookings.API/Controllers/HearingsController.cs

```diff
@@ -116,71 +125,128 @@ public async Task<IActionResult> GetHearingsByUsername([FromQuery]string usernam
         [ProducesResponseType((int)HttpStatusCode.BadRequest)]
         public async Task<IActionResult> BookNewHearing(BookNewHearingRequest request)
         {
-            var result = new BookNewHearingRequestValidation().Validate(request);
-            if (!result.IsValid)
+            try
             {
-                ModelState.AddFluentValidationErrors(result.Errors);
-                return BadRequest(ModelState);
-            }
-
-            var query = new GetCaseTypeQuery(request.CaseTypeName);
-            var caseType = await _queryHandler.Handle<GetCaseTypeQuery, CaseType>(query);
-
+                if (request == null)
+                {
+                    ModelState.AddModelError(nameof(BookNewHearingRequest), "BookNewHearingRequest is null");
+                    _logger.TrackEvent("BookNewHearing Error: BookNewHearingRequest is null");
+                    return BadRequest(ModelState);
+                }
+                
+                var result = await new BookNewHearingRequestValidation().ValidateAsync(request);
+                if (!result.IsValid)
+                {
+                    ModelState.AddFluentValidationErrors(result.Errors);
+                    var dictionary = result.Errors.ToDictionary(x => $"{x.PropertyName}-{Guid.NewGuid()}", x => x.ErrorMessage);
+                    var payload = JsonConvert.SerializeObject(request);
+                    dictionary.Add("payload", !string.IsNullOrWhiteSpace(payload) ? payload : "Empty Payload");
+                    _logger.TrackEvent("BookNewHearing Validation Errors", dictionary);
+                    return BadRequest(ModelState);
+                }
 
-            if (caseType == null)
-            {
-                ModelState.AddModelError(nameof(request.CaseTypeName), "Case type does not exist");
-                return BadRequest(ModelState);
-            }
+                var query = new GetCaseTypeQuery(request.CaseTypeName);
+                var caseType = await _queryHandler.Handle<GetCaseTypeQuery, CaseType>(query);
+                
+                if (caseType == null)
+                {
+                    ModelState.AddModelError(nameof(request.CaseTypeName), "Case type does not exist");
+                    _logger.TrackEvent("BookNewHearing Error: Case type does not exist", new Dictionary<string, string>{{"CaseTypeName", request?.CaseTypeName}});
```

Comment:
could we replace trackevent to logger.logdebug or logger.logwarning so the logs can fall under the traces section

Reply:
Good spot Shaed. I agree.

Reply:
Yeh, I decided to do TrackEvent so we can quickly see the info without going through all the traces